File: go/src/sorts/insertionsort_test.go
package sorts

import "../testhelper"

import "testing"

func InsertionSortTest(t *testing.T) {
  arr := testhelper.RandomIntSlice(testhelper.ArraySize)

  sorted := InsertionSort(arr)

  testhelper.AssertIsSorted(sorted, t)
}

File: python/structures/test/sorts/insertion_sort_test.py
import structures.src.sorts.insertion as insertion_sort
from structures.src.util.sort_util import random_array, is_sorted
import structures.src.util.constants as const


def test_sorted():
    arr = random_array(const.ARRAY_SIZE)

    # make sure the starting array isn't sorted
    assert not is_sorted(arr)

    # sort the array
    result = insertion_sort.sort(arr)

    # make sure it worked
    assert is_sorted(result)
